Plugins/NativeCamera/Source/NativeVideo.cpp

```diff
@@ -0,0 +1,156 @@
+                InstanceMethod("pause", &NativeVideo::Pause),
+                InstanceMethod("setAttribute", &NativeVideo::SetAttribute),
+                InstanceAccessor("videoWidth", &NativeVideo::GetVideoWidth, &NativeVideo::SetVideoWidth),
+                InstanceAccessor("videoHeight", &NativeVideo::GetVideoHeight, &NativeVideo::SetVideoHeight),
+                InstanceAccessor("frontCamera", nullptr, &NativeVideo::SetFrontCamera),
+                InstanceAccessor("isNative", &NativeVideo::IsNative, nullptr),
+                InstanceAccessor("readyState", &NativeVideo::GetReadyState, nullptr),
+                InstanceAccessor("HAVE_CURRENT_DATA", &NativeVideo::GetHaveCurrentData, nullptr),
+            });
+
+        env.Global().Set(JS_CLASS_NAME, func);
+    }
+
+    Napi::Object NativeVideo::New(const Napi::CallbackInfo& info, uint32_t width, uint32_t height, bool frontCamera)
+    {
+        return info.Env().Global().Get(JS_CLASS_NAME).As<Napi::Function>().New({ Napi::Value::From(info.Env(), width), Napi::Value::From(info.Env(), height), Napi::Value::From(info.Env(), frontCamera) });
+    }
+
+    NativeVideo::NativeVideo(const Napi::CallbackInfo& info)
+        : Napi::ObjectWrap<NativeVideo>{ info }
+        , m_width{ info[0].As<Napi::Number>().Uint32Value() }
+        , m_height{ info[1].As<Napi::Number>().Uint32Value() }
+        , m_frontCamera{ info[2].As<Napi::Boolean>().Value() }
+    {
+    }
+
+    void NativeVideo::SetVideoWidth(const Napi::CallbackInfo&, const Napi::Value& value)
+    {
+        m_width = value.As<Napi::Number>().Uint32Value();
+    }
+
+    Napi::Value NativeVideo::GetVideoWidth(const Napi::CallbackInfo& /*info*/)
+    {
+        return Napi::Value::From(Env(), m_width);
+    }
+
+    void NativeVideo::SetVideoHeight(const Napi::CallbackInfo&, const Napi::Value& value)
+    {
+        m_height = value.As<Napi::Number>().Uint32Value();
+    }
+
+    Napi::Value NativeVideo::GetVideoHeight(const Napi::CallbackInfo& /*info*/)
+    {
+        return Napi::Value::From(Env(), m_height);
+    }
+
+    void NativeVideo::SetFrontCamera(const Napi::CallbackInfo&, const Napi::Value& value)
+    {
+        m_frontCamera = value.As<Napi::Boolean>().Value();
+    }
+
+    Napi::Value NativeVideo::IsNative(const Napi::CallbackInfo&)
+    {
+        return Napi::Value::From(Env(), true);
+    }
+
+    void NativeVideo::SetAttribute(const Napi::CallbackInfo&)
+    {
+    }
+
+    Napi::Value NativeVideo::GetReadyState(const Napi::CallbackInfo& /*info*/)
+    {
+        return Napi::Value::From(Env(), 10u);
+    }
+
+    Napi::Value NativeVideo::GetHaveCurrentData(const Napi::CallbackInfo& /*info*/)
+    {
+        return Napi::Value::From(Env(), 1u);
+    }
+
+    void NativeVideo::UpdateTexture(bgfx::TextureHandle textureHandle)
+    {
+        if (!m_cameraInterface)
+        {
+            m_cameraInterface = CameraInterface::CreateInterface(Env(), m_width, m_height, m_frontCamera);
```

Comment:
From what I can tell, the Android and iOS implementations both create the camera interface asynchronously, but then we call UpdateCameraTexture immediately after this. How is this supposed to work? Is UpdateCameraTexture just supposed to be a no-op during async creation?